### ISSUE TITLE
Dimmed background blurring when switching between special workspaces

### ISSUE THREAD

**Issue by BRS5672023** [hyprwm/Hyprland#5243]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch HEAD at commit c5e28ebcfe00a510922779b2c568cfa52a317445  (props: bump ver 0.37.1).
Date: Sat Mar 16 14:51:49 2024
Tag: v0.37.1

flags: (if any)


System Information:
System name: Linux
Node name: D30
Release: 6.8.1-zen1-1-zen
Version: #1 ZEN SMP PREEMPT_DYNAMIC Sat, 16 Mar 2024 17:15:23 +0000


GPU information: 
09:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Navi 32 [Radeon RX 7700 XT / 7800 XT] [1002:747e] (rev c8) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Bug or Regression?

Regression

### Description

When switching between special workspaces, the blurred background seems dimmed a bit at first (which happens even when I set `dim_special=0` in hyprland.conf) while toggle an existing one won't have this issue.

This issue occurs since version 0.33.0-1.

### How to reproduce

Open multiple special workspaces, and switching between them.

### Crash reports, logs, images, videos

https://github.com/hyprwm/Hyprland/assets/162549262/f505dd20-3661-47a4-9ffd-86312f0191f4

**Comment by vaxerski** [comment]
does it happen without blur too?

**Comment by BRS5672023** [comment]
> does it happen without blur too?

Yes, the same issue occurs and maybe more evident if no blur but transparency enabled

**Comment by vaxerski** [comment]
hm, sounds like double rendering. Is this a regression? does this happen in e.g. 0.34?

**Comment by BRS5672023** [comment]
> hm, sounds like double rendering. Is this a regression? does this happen in e.g. 0.34?

Okay, it seems to be a regression, happens on 0.33.0-1, but not on 0.32.3-1

**Comment by BRS5672023** [comment]
> could you git bisect?

Sorry that I'm not familiar with git, please tell me anything so that I can help...

**Comment by thejch** [comment]
Seems to be from the same time when the other bunch of double rendering issues happened. It was actually mentioned before here: https://github.com/hyprwm/Hyprland/issues/3888#issuecomment-1826900936

**Comment by vaxerski** [comment]
forgot about that, heh. Wanna fix or should I do it

**Comment by thejch** [comment]
you fix pls

**Comment by SleeperOfSaturn** [comment]
Bruh use git bisect

**Comment by vaxerski** [comment]
no need

**Comment by KAGEYAM4** [comment]
Another way to reproduce - 

```
Hyprland, built from branch HEAD at commit 3875679755014997776e091ff8903acfb311dd2f  (props: bump ver to 0.38.0).
Date: Mon Apr 1 18:30:37 2024
Tag: v0.38.0
```

[Kooha-2024-04-03-16-00-41.webm](https://github.com/hyprwm/Hyprland/assets/75798544/082566a0-f06b-4a36-96a0-2bfca5c58340)

**Comment by KAGEYAM4** [comment]
@BRS5672023  after 0.38.1, does this bug still happen for you? you can close the issue if it dosen't.

**Comment by BRS5672023** [comment]
> @BRS5672023 after 0.38.1, does this bug still happen for you? you can close the issue if it dosen't.

Yes, it still happens to me...

**Comment by KAGEYAM4** [comment]
> Yes, it still happens to me...

I downgraded to 38.0 and tried reproducing your bug ->
```
bind = $mainMod, S, togglespecialworkspace,
bind = $mainMod, R, togglespecialworkspace , what
```
Opened floating terminal on one and anki on other, switched between them. could reproduce what you have shown in the video.

I was able to reproduce the bug i posted above.

Not sure if the bug i was facing the same thing or not.


--------

You could try to see if you can reproduce same thing with clean hyprland config with enough changes to reproduce the bug.

**Comment by BRS5672023** [comment]
> > Yes, it still happens to me...
> 
> I downgraded to 38.0 and tried reproducing your bug ->
> 
> ```
> bind = $mainMod, S, togglespecialworkspace,
> bind = $mainMod, R, togglespecialworkspace , what
> ```
> 
> Opened floating terminal on one and anki on other, switched between them. but could not reproduce what you have shown in the video.
> 
> I was able to reproduce the bug i posted above.
> 
> Not sure if the bug i was facing the same thing or not.
> 
> You could try to see if you can reproduce same thing with clean hyprland config with enough changes to reproduce the bug.

I'm not sure what do you mean for "floating", you mean floating in a special workspace?
```
workspace=special:dropterm,on-created-empty:kitty
workspace=special:dropterm1,on-created-empty:kitty

bind=$mainMod,T,togglespecialworkspace,dropterm
bind=$mainMod,M,togglespecialworkspace,dropterm1
```
With lines above, I could reproduce this bug on version 0.38.1-1 by alternatively toggling between two special workspaces.

**Comment by KAGEYAM4** [comment]
> workspace=special:dropterm,on-created-empty:kitty
> workspace=special:dropterm1,on-created-empty:kitty
> 
> bind=$mainMod,T,togglespecialworkspace,dropterm
> bind=$mainMod,M,togglespecialworkspace,dropterm1
> ```
> 
> With lines above, I could reproduce this bug on version 0.38.1-1 by alternatively toggling between two special workspaces.

ok, i could reproduce it too now, it wasn't working before because i had `xray = true # blur section`.

**Comment by izmyname** [comment]
Is it still a thing?

**Comment by BRS5672023** [comment]
> Is it still a thing?

I can still repro it on AQ

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
